docs/glossary.md

```diff
@@ -0,0 +1,258 @@
+application-specific functionality.
+
+**Transaction** - A data storage unit on Arweave identified by a unique
+43-character base64url ID. Contains data, tags, and cryptographic signatures
+proving ownership and integrity.
+
+## ANS-104 Bundle System
+
+**ANS-102/ANS-104** - Arweave standards for bundling multiple data items into a
+single transaction. ANS-104 is the current standard supporting advanced features
+like nested bundles.
+
+**BDI (Bundle Data Item)** - A data item that itself is a bundle, enabling
+nested bundle structures within the ANS-104 standard.
+
+**Bundle** - A collection of data items packed into a single Arweave
+transaction. Bundles reduce costs and improve throughput by amortizing
+transaction overhead.
+
+**Bundle Format** - The specific standard (ANS-102 or ANS-104) used to encode a
+bundle. The format determines how data items are structured and indexed within
+the bundle.
+
+**Data Item** - An individual piece of data within a bundle. Each data item has
+its own ID, signature, tags, and data, similar to a transaction but more
+lightweight.
+
+**Data Item ID** - A unique identifier for a data item within a bundle,
+calculated from the data item's signature.
+
+**Nested Bundle** - A bundle contained within another bundle. Identified by
+having a parent_id field pointing to the containing bundle.
+
+**Parent ID** - The identifier of the containing bundle or transaction for a
+data item. For data items in nested bundles, points to the immediate parent
+bundle. For top-level data items, equals the root transaction ID.
+
+**Root Transaction ID** - The top-level Arweave transaction ID containing a
+bundle or data item. Used to retrieve the raw data from Arweave.
+
+**Unbundling** - The process of extracting and indexing individual data items
+from a bundle transaction.
+
+## AR.IO Gateway Concepts
+
+**ArNS (Arweave Name System)** - A decentralized naming system that maps
+human-readable names to Arweave transaction IDs, similar to DNS for the
+permaweb.
+
+**Contiguous Data** - Complete, reassembled data from all chunks of a
+transaction. The gateway stores metadata about contiguous data availability and
+verification status.
+
+**Data Verification** - The process of cryptographically verifying that
+retrieved data matches its data root hash, ensuring data integrity.
+
+**Observer** - A gateway component that monitors and reports on the health and
+behavior of other gateways in the AR.IO network.
+
+**X-AR-IO Headers** - HTTP headers added by the gateway providing metadata about
+data verification, caching, and trust status.
+
+**X-ArNS Headers** - HTTP response headers added during ArNS name resolution
+that provide metadata about the resolution process, including the resolved
+transaction ID, TTL, process ID, and undername information.
+
+## Databases
+
+**Bundles Database** - SQLite database storing ANS-104 bundle metadata, data
+items, and their relationships. Includes retry tracking for bundle processing.
```

Comment:
Nit: might want to add "data item tags" explicitly. As currently written, that's implied, but perhaps not clear to the uninitiated.